### PR TITLE
fix(requests): ensure http metadata

### PR DESCRIPTION
**Pull request by P403n1x87**
## Description

Call `set_http_meta` once and with all the available data. Also avoid `urlparse` and use simpler parsing instead for better performance.

## Checklist
- [ ] Added to the correct milestone.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

**Comment by brettlangdon** [comment]
Do you have any benchmarks/performance numbers for these changes?

File: ddtrace/contrib/requests/connection.py [hunk 1]
```diff
@@ -29,10 +28,14 @@ def _wrap_send(func, instance, args, kwargs):
...
         return func(*args, **kwargs)
 
-    parsed_uri = parse.urlparse(request.url)
-    hostname = parsed_uri.hostname
-    if parsed_uri.port:
-        hostname = "{}:{}".format(hostname, parsed_uri.port)
+    url = request.url
+    i = url.index("://") + 3
+    try:
+        j = url.index("/", i)
```
Comment (nizox) [inline ddtrace/contrib/requests/connection.py:34]: `http://example.com?foo` and `http://example.com#foo` are valid URLs. We probably need to handle them too.

**Review by nizox — CHANGES_REQUESTED** [review]
Small comment otherwise LGTM

File: ddtrace/contrib/requests/connection.py (hunk 1 above)
Comment (P403n1x87) [reply ddtrace/contrib/requests/connection.py:34]: Indeed! Good catch!

**Review by brettlangdon — APPROVED** [review]
lgtm

**Review by Yun-Kim — APPROVED** [review]

**Pull request merged**